test/tools/deprecate_warning_test_program.js

```diff
@@ -0,0 +1,27 @@
+'use strict';
+
+if (require.main !== module) {
```

Comment:
Add a comment explaining why we are doing this (so this file cannot be imported by anyone)